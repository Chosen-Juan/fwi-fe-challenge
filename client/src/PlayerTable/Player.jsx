import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Flags from 'react-world-flags';
import { connect } from 'react-redux'

import Avatar from '../Avatar';
import { COUNTRIES, COUNTRIES_ARRAY } from '../constants';
import { editPlayer, deletePlayer } from '../appState/actions';

export class Player extends Component {
  static propTypes = {
    player: PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      country: PropTypes.oneOf(Object.keys(COUNTRIES)),
      winnings: PropTypes.number.isRequired,
      imageUrl: PropTypes.string.isRequired,
    }),
    editPlayer: PropTypes.func.isRequired,
    deletePlayer: PropTypes.func.isRequired,
  };

  handleButtonPress = () => {
    this.buttonPressTimer = setTimeout(() => this.setEditing(true), 1000);
  };

  handleButtonRelease = () => {
    clearTimeout(this.buttonPressTimer);
  };

  state = {
    editing: false,
    name: this.props.player.name,
    winnings: this.props.player.winnings,
    country: this.props.player.country,
  };

  setEditing = editing => {
    this.setState({ editing });
  };

  deletePlayer = () => {
    const { deletePlayer, player } = this.props;
    deletePlayer(player.id);
  };

  saveEdit = () => {
    const { id } = this.props.player;
    const { name, winnings, country } = this.state;
    this.props.editPlayer(id, name, parseInt(winnings), country)
      .then(this.finishEditing);
  };

  finishEditing = () => {
    this.setState({ success: true });
    setTimeout(() => {
      this.setState({ success: false });
      this.setEditing(false);
    }, 3000);
  };

  onChangeHandler = ({ target }) => {
    this.setState({ [target.name]: target.value });
  };

  renderCountryOptions = () => {
    const options = [
      <option key="default" disabled value>Select a country</option>
    ];
    COUNTRIES_ARRAY.forEach(country => {
      options.push(
        <option key={country.code} value={country.code}>
          {country.value}
        </option>
      );
    });
    return options;
  }

  render() {
    const { player } = this.props;
    return (
      <tr
        key={player.id}
        role="row"
        className="table__row"
        onTouchStart={this.handleButtonPress}
        onTouchEnd={this.handleButtonRelease}
      >
        <td className={this.state.editing ? "is-hidden-desktop" : "is-hidden"}>
          <button className="button is-primary" onClick={this.saveEdit}>Save</button>
          <button className="button is-danger" onClick={() => this.setEditing(false)}>Cancel</button>
        </td>
        <td className="is-narrow">
          <Avatar src={player.imageUrl} />
        </td>
        <td hidden={this.state.editing}>
          {player.name}
        </td>
        <td hidden={!this.state.editing}>
          <input name="name" type="text" className="input" value={this.state.name} onChange={this.onChangeHandler} />
        </td>
        <td hidden={this.state.editing}>
          {player.winnings.toLocaleString(undefined, {
            style: 'currency',
            currency: 'USD',
          })}
        </td>
        <td hidden={!this.state.editing}>
          <input name="winnings" type="number" className="input" value={this.state.winnings} onChange={this.onChangeHandler} />
        </td>
        <td hidden={this.state.editing}>
          <div className="country">
            <Avatar>
              <Flags code={player.country} alt="" />
            </Avatar>
            {player.country}
          </div>
        </td>
        <td hidden={!this.state.editing}>
          <select name="country" className="input" value={this.state.country} onChange={this.onChangeHandler}>
            {this.renderCountryOptions()}
          </select>
        </td>
        <td className={this.state.success ? "notification is-success" : "is-hidden"}>Saved</td>
        <td className={this.state.success ? "is-hidden" : "is-narrow"}>
          <div className={this.state.editing ? "is-hidden-touch": "is-hidden"}>
            <button className="button is-primary" onClick={this.saveEdit}>Save</button>
            <button className="button is-danger" onClick={() => this.setEditing(false)}>Cancel</button>
          </div>
          <button className={this.state.editing ? "is-hidden" : "button is-link is-outlined is-hidden-touch"} onClick={() => this.setEditing(true)}>Edit</button>
          <button className={this.state.editing ? "is-hidden" : "button is-danger is-outlined"} onClick={this.deletePlayer}>&times;</button>
        </td>
      </tr>
    );
  }
}
const mapStateToProps = state => ({
  playerStore: state.player
});

const mapDispatchToProps = {
  editPlayer, deletePlayer
};

export default connect(mapStateToProps, mapDispatchToProps)(Player);
