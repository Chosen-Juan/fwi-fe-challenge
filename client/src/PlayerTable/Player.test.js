import React from 'react';
import { shallow } from 'enzyme';

import { Player } from './Player';

describe('Player', () => {
  const editPlayerMock = jest.fn(() => Promise.resolve());
  const deletePlayerMock = jest.fn();
  const defaultProps = {
    player: {
      id: '1one',
      name: 'Jon Doe',
      country: 'US',
      winnings: 24999,
      imageUrl: ''
    },
    editPlayer: editPlayerMock,
    deletePlayer: deletePlayerMock,
  };
  const player = shallow(<Player {...defaultProps} />);

  it('should match snapshot', () => {
    expect(player).toMatchSnapshot();
  });

  it('should set editing on and off', () => {
    const editButton = player.find('.button.is-link');
    const cancelButton = player.find('.button.is-danger').at(0);
    editButton.simulate('click');
    expect(player.state('editing')).toEqual(true);
    cancelButton.simulate('click');
    expect(player.state('editing')).toEqual(false);
  });

  it('should call edit player', () => {
    const saveButton = player.find('.button.is-primary').at(1);
    saveButton.simulate('click');
    expect(editPlayerMock).toHaveBeenCalledTimes(1);
  });

});
