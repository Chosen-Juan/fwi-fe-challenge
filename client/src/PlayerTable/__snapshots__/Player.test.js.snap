// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player should match snapshot 1`] = `
<tr
  className="table__row"
  key="1one"
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  role="row"
>
  <td
    className="is-hidden"
  >
    <button
      className="button is-primary"
      onClick={[Function]}
    >
      Save
    </button>
    <button
      className="button is-danger"
      onClick={[Function]}
    >
      Cancel
    </button>
  </td>
  <td
    className="is-narrow"
  >
    <Avatar
      src=""
    />
  </td>
  <td
    hidden={false}
  >
    Jon Doe
  </td>
  <td
    hidden={true}
  >
    <input
      className="input"
      name="name"
      onChange={[Function]}
      type="text"
      value="Jon Doe"
    />
  </td>
  <td
    hidden={false}
  >
    US$Â 24,999.00
  </td>
  <td
    hidden={true}
  >
    <input
      className="input"
      name="winnings"
      onChange={[Function]}
      type="number"
      value={24999}
    />
  </td>
  <td
    hidden={false}
  >
    <div
      className="country"
    >
      <Avatar>
        <Component
          alt=""
          code="US"
        />
      </Avatar>
      US
    </div>
  </td>
  <td
    hidden={true}
  >
    <select
      className="input"
      name="country"
      onChange={[Function]}
      value="US"
    >
      <option
        disabled={true}
        key="default"
        value={true}
      >
        Select a country
      </option>
      <option
        key="AF"
        value="AF"
      >
        Afghanistan
      </option>
      <option
        key="AX"
        value="AX"
      >
        Aland Islands
      </option>
      <option
        key="AL"
        value="AL"
      >
        Albania
      </option>
      <option
        key="DZ"
        value="DZ"
      >
        Algeria
      </option>
      <option
        key="AS"
        value="AS"
      >
        American Samoa
      </option>
      <option
        key="AD"
        value="AD"
      >
        Andorra
      </option>
      <option
        key="AO"
        value="AO"
      >
        Angola
      </option>
      <option
        key="AI"
        value="AI"
      >
        Anguilla
      </option>
      <option
        key="AQ"
        value="AQ"
      >
        Antarctica
      </option>
      <option
        key="AG"
        value="AG"
      >
        Antigua and Barbuda
      </option>
      <option
        key="AR"
        value="AR"
      >
        Argentina
      </option>
      <option
        key="AM"
        value="AM"
      >
        Armenia
      </option>
      <option
        key="AW"
        value="AW"
      >
        Aruba
      </option>
      <option
        key="AU"
        value="AU"
      >
        Australia
      </option>
      <option
        key="AT"
        value="AT"
      >
        Austria
      </option>
      <option
        key="AZ"
        value="AZ"
      >
        Azerbaijan
      </option>
      <option
        key="BS"
        value="BS"
      >
        Bahamas
      </option>
      <option
        key="BH"
        value="BH"
      >
        Bahrain
      </option>
      <option
        key="BD"
        value="BD"
      >
        Bangladesh
      </option>
      <option
        key="BB"
        value="BB"
      >
        Barbados
      </option>
      <option
        key="BY"
        value="BY"
      >
        Belarus
      </option>
      <option
        key="BE"
        value="BE"
      >
        Belgium
      </option>
      <option
        key="BZ"
        value="BZ"
      >
        Belize
      </option>
      <option
        key="BJ"
        value="BJ"
      >
        Benin
      </option>
      <option
        key="BM"
        value="BM"
      >
        Bermuda
      </option>
      <option
        key="BT"
        value="BT"
      >
        Bhutan
      </option>
      <option
        key="BO"
        value="BO"
      >
        Bolivia
      </option>
      <option
        key="BQ"
        value="BQ"
      >
        Bonaire, Saint Eustatius and Saba 
      </option>
      <option
        key="BA"
        value="BA"
      >
        Bosnia and Herzegovina
      </option>
      <option
        key="BW"
        value="BW"
      >
        Botswana
      </option>
      <option
        key="BV"
        value="BV"
      >
        Bouvet Island
      </option>
      <option
        key="BR"
        value="BR"
      >
        Brazil
      </option>
      <option
        key="IO"
        value="IO"
      >
        British Indian Ocean Territory
      </option>
      <option
        key="VG"
        value="VG"
      >
        British Virgin Islands
      </option>
      <option
        key="BN"
        value="BN"
      >
        Brunei
      </option>
      <option
        key="BG"
        value="BG"
      >
        Bulgaria
      </option>
      <option
        key="BF"
        value="BF"
      >
        Burkina Faso
      </option>
      <option
        key="BI"
        value="BI"
      >
        Burundi
      </option>
      <option
        key="KH"
        value="KH"
      >
        Cambodia
      </option>
      <option
        key="CM"
        value="CM"
      >
        Cameroon
      </option>
      <option
        key="CA"
        value="CA"
      >
        Canada
      </option>
      <option
        key="CV"
        value="CV"
      >
        Cape Verde
      </option>
      <option
        key="KY"
        value="KY"
      >
        Cayman Islands
      </option>
      <option
        key="CF"
        value="CF"
      >
        Central African Republic
      </option>
      <option
        key="TD"
        value="TD"
      >
        Chad
      </option>
      <option
        key="CL"
        value="CL"
      >
        Chile
      </option>
      <option
        key="CN"
        value="CN"
      >
        China
      </option>
      <option
        key="CX"
        value="CX"
      >
        Christmas Island
      </option>
      <option
        key="CC"
        value="CC"
      >
        Cocos Islands
      </option>
      <option
        key="CO"
        value="CO"
      >
        Colombia
      </option>
      <option
        key="KM"
        value="KM"
      >
        Comoros
      </option>
      <option
        key="CK"
        value="CK"
      >
        Cook Islands
      </option>
      <option
        key="CR"
        value="CR"
      >
        Costa Rica
      </option>
      <option
        key="HR"
        value="HR"
      >
        Croatia
      </option>
      <option
        key="CU"
        value="CU"
      >
        Cuba
      </option>
      <option
        key="CW"
        value="CW"
      >
        Curacao
      </option>
      <option
        key="CY"
        value="CY"
      >
        Cyprus
      </option>
      <option
        key="CZ"
        value="CZ"
      >
        Czech Republic
      </option>
      <option
        key="CD"
        value="CD"
      >
        Democratic Republic of the Congo
      </option>
      <option
        key="DK"
        value="DK"
      >
        Denmark
      </option>
      <option
        key="DJ"
        value="DJ"
      >
        Djibouti
      </option>
      <option
        key="DM"
        value="DM"
      >
        Dominica
      </option>
      <option
        key="DO"
        value="DO"
      >
        Dominican Republic
      </option>
      <option
        key="TL"
        value="TL"
      >
        East Timor
      </option>
      <option
        key="EC"
        value="EC"
      >
        Ecuador
      </option>
      <option
        key="EG"
        value="EG"
      >
        Egypt
      </option>
      <option
        key="SV"
        value="SV"
      >
        El Salvador
      </option>
      <option
        key="GQ"
        value="GQ"
      >
        Equatorial Guinea
      </option>
      <option
        key="ER"
        value="ER"
      >
        Eritrea
      </option>
      <option
        key="EE"
        value="EE"
      >
        Estonia
      </option>
      <option
        key="ET"
        value="ET"
      >
        Ethiopia
      </option>
      <option
        key="FK"
        value="FK"
      >
        Falkland Islands
      </option>
      <option
        key="FO"
        value="FO"
      >
        Faroe Islands
      </option>
      <option
        key="FJ"
        value="FJ"
      >
        Fiji
      </option>
      <option
        key="FI"
        value="FI"
      >
        Finland
      </option>
      <option
        key="FR"
        value="FR"
      >
        France
      </option>
      <option
        key="GF"
        value="GF"
      >
        French Guiana
      </option>
      <option
        key="PF"
        value="PF"
      >
        French Polynesia
      </option>
      <option
        key="TF"
        value="TF"
      >
        French Southern Territories
      </option>
      <option
        key="GA"
        value="GA"
      >
        Gabon
      </option>
      <option
        key="GM"
        value="GM"
      >
        Gambia
      </option>
      <option
        key="GE"
        value="GE"
      >
        Georgia
      </option>
      <option
        key="DE"
        value="DE"
      >
        Germany
      </option>
      <option
        key="GH"
        value="GH"
      >
        Ghana
      </option>
      <option
        key="GI"
        value="GI"
      >
        Gibraltar
      </option>
      <option
        key="GR"
        value="GR"
      >
        Greece
      </option>
      <option
        key="GL"
        value="GL"
      >
        Greenland
      </option>
      <option
        key="GD"
        value="GD"
      >
        Grenada
      </option>
      <option
        key="GP"
        value="GP"
      >
        Guadeloupe
      </option>
      <option
        key="GU"
        value="GU"
      >
        Guam
      </option>
      <option
        key="GT"
        value="GT"
      >
        Guatemala
      </option>
      <option
        key="GG"
        value="GG"
      >
        Guernsey
      </option>
      <option
        key="GN"
        value="GN"
      >
        Guinea
      </option>
      <option
        key="GW"
        value="GW"
      >
        Guinea-Bissau
      </option>
      <option
        key="GY"
        value="GY"
      >
        Guyana
      </option>
      <option
        key="HT"
        value="HT"
      >
        Haiti
      </option>
      <option
        key="HM"
        value="HM"
      >
        Heard Island and McDonald Islands
      </option>
      <option
        key="HN"
        value="HN"
      >
        Honduras
      </option>
      <option
        key="HK"
        value="HK"
      >
        Hong Kong
      </option>
      <option
        key="HU"
        value="HU"
      >
        Hungary
      </option>
      <option
        key="IS"
        value="IS"
      >
        Iceland
      </option>
      <option
        key="IN"
        value="IN"
      >
        India
      </option>
      <option
        key="ID"
        value="ID"
      >
        Indonesia
      </option>
      <option
        key="IR"
        value="IR"
      >
        Iran
      </option>
      <option
        key="IQ"
        value="IQ"
      >
        Iraq
      </option>
      <option
        key="IE"
        value="IE"
      >
        Ireland
      </option>
      <option
        key="IM"
        value="IM"
      >
        Isle of Man
      </option>
      <option
        key="IL"
        value="IL"
      >
        Israel
      </option>
      <option
        key="IT"
        value="IT"
      >
        Italy
      </option>
      <option
        key="CI"
        value="CI"
      >
        Ivory Coast
      </option>
      <option
        key="JM"
        value="JM"
      >
        Jamaica
      </option>
      <option
        key="JP"
        value="JP"
      >
        Japan
      </option>
      <option
        key="JE"
        value="JE"
      >
        Jersey
      </option>
      <option
        key="JO"
        value="JO"
      >
        Jordan
      </option>
      <option
        key="KZ"
        value="KZ"
      >
        Kazakhstan
      </option>
      <option
        key="KE"
        value="KE"
      >
        Kenya
      </option>
      <option
        key="KI"
        value="KI"
      >
        Kiribati
      </option>
      <option
        key="XK"
        value="XK"
      >
        Kosovo
      </option>
      <option
        key="KW"
        value="KW"
      >
        Kuwait
      </option>
      <option
        key="KG"
        value="KG"
      >
        Kyrgyzstan
      </option>
      <option
        key="LA"
        value="LA"
      >
        Laos
      </option>
      <option
        key="LV"
        value="LV"
      >
        Latvia
      </option>
      <option
        key="LB"
        value="LB"
      >
        Lebanon
      </option>
      <option
        key="LS"
        value="LS"
      >
        Lesotho
      </option>
      <option
        key="LR"
        value="LR"
      >
        Liberia
      </option>
      <option
        key="LY"
        value="LY"
      >
        Libya
      </option>
      <option
        key="LI"
        value="LI"
      >
        Liechtenstein
      </option>
      <option
        key="LT"
        value="LT"
      >
        Lithuania
      </option>
      <option
        key="LU"
        value="LU"
      >
        Luxembourg
      </option>
      <option
        key="MO"
        value="MO"
      >
        Macao
      </option>
      <option
        key="MK"
        value="MK"
      >
        Macedonia
      </option>
      <option
        key="MG"
        value="MG"
      >
        Madagascar
      </option>
      <option
        key="MW"
        value="MW"
      >
        Malawi
      </option>
      <option
        key="MY"
        value="MY"
      >
        Malaysia
      </option>
      <option
        key="MV"
        value="MV"
      >
        Maldives
      </option>
      <option
        key="ML"
        value="ML"
      >
        Mali
      </option>
      <option
        key="MT"
        value="MT"
      >
        Malta
      </option>
      <option
        key="MH"
        value="MH"
      >
        Marshall Islands
      </option>
      <option
        key="MQ"
        value="MQ"
      >
        Martinique
      </option>
      <option
        key="MR"
        value="MR"
      >
        Mauritania
      </option>
      <option
        key="MU"
        value="MU"
      >
        Mauritius
      </option>
      <option
        key="YT"
        value="YT"
      >
        Mayotte
      </option>
      <option
        key="MX"
        value="MX"
      >
        Mexico
      </option>
      <option
        key="FM"
        value="FM"
      >
        Micronesia
      </option>
      <option
        key="MD"
        value="MD"
      >
        Moldova
      </option>
      <option
        key="MC"
        value="MC"
      >
        Monaco
      </option>
      <option
        key="MN"
        value="MN"
      >
        Mongolia
      </option>
      <option
        key="ME"
        value="ME"
      >
        Montenegro
      </option>
      <option
        key="MS"
        value="MS"
      >
        Montserrat
      </option>
      <option
        key="MA"
        value="MA"
      >
        Morocco
      </option>
      <option
        key="MZ"
        value="MZ"
      >
        Mozambique
      </option>
      <option
        key="MM"
        value="MM"
      >
        Myanmar
      </option>
      <option
        key="NA"
        value="NA"
      >
        Namibia
      </option>
      <option
        key="NR"
        value="NR"
      >
        Nauru
      </option>
      <option
        key="NP"
        value="NP"
      >
        Nepal
      </option>
      <option
        key="NL"
        value="NL"
      >
        Netherlands
      </option>
      <option
        key="NC"
        value="NC"
      >
        New Caledonia
      </option>
      <option
        key="NZ"
        value="NZ"
      >
        New Zealand
      </option>
      <option
        key="NI"
        value="NI"
      >
        Nicaragua
      </option>
      <option
        key="NE"
        value="NE"
      >
        Niger
      </option>
      <option
        key="NG"
        value="NG"
      >
        Nigeria
      </option>
      <option
        key="NU"
        value="NU"
      >
        Niue
      </option>
      <option
        key="NF"
        value="NF"
      >
        Norfolk Island
      </option>
      <option
        key="KP"
        value="KP"
      >
        North Korea
      </option>
      <option
        key="MP"
        value="MP"
      >
        Northern Mariana Islands
      </option>
      <option
        key="NO"
        value="NO"
      >
        Norway
      </option>
      <option
        key="OM"
        value="OM"
      >
        Oman
      </option>
      <option
        key="PK"
        value="PK"
      >
        Pakistan
      </option>
      <option
        key="PW"
        value="PW"
      >
        Palau
      </option>
      <option
        key="PS"
        value="PS"
      >
        Palestinian Territory
      </option>
      <option
        key="PA"
        value="PA"
      >
        Panama
      </option>
      <option
        key="PG"
        value="PG"
      >
        Papua New Guinea
      </option>
      <option
        key="PY"
        value="PY"
      >
        Paraguay
      </option>
      <option
        key="PE"
        value="PE"
      >
        Peru
      </option>
      <option
        key="PH"
        value="PH"
      >
        Philippines
      </option>
      <option
        key="PN"
        value="PN"
      >
        Pitcairn
      </option>
      <option
        key="PL"
        value="PL"
      >
        Poland
      </option>
      <option
        key="PT"
        value="PT"
      >
        Portugal
      </option>
      <option
        key="PR"
        value="PR"
      >
        Puerto Rico
      </option>
      <option
        key="QA"
        value="QA"
      >
        Qatar
      </option>
      <option
        key="CG"
        value="CG"
      >
        Republic of the Congo
      </option>
      <option
        key="RE"
        value="RE"
      >
        Reunion
      </option>
      <option
        key="RO"
        value="RO"
      >
        Romania
      </option>
      <option
        key="RU"
        value="RU"
      >
        Russia
      </option>
      <option
        key="RW"
        value="RW"
      >
        Rwanda
      </option>
      <option
        key="BL"
        value="BL"
      >
        Saint Barthelemy
      </option>
      <option
        key="SH"
        value="SH"
      >
        Saint Helena
      </option>
      <option
        key="KN"
        value="KN"
      >
        Saint Kitts and Nevis
      </option>
      <option
        key="LC"
        value="LC"
      >
        Saint Lucia
      </option>
      <option
        key="MF"
        value="MF"
      >
        Saint Martin
      </option>
      <option
        key="PM"
        value="PM"
      >
        Saint Pierre and Miquelon
      </option>
      <option
        key="VC"
        value="VC"
      >
        Saint Vincent and the Grenadines
      </option>
      <option
        key="WS"
        value="WS"
      >
        Samoa
      </option>
      <option
        key="SM"
        value="SM"
      >
        San Marino
      </option>
      <option
        key="ST"
        value="ST"
      >
        Sao Tome and Principe
      </option>
      <option
        key="SA"
        value="SA"
      >
        Saudi Arabia
      </option>
      <option
        key="SN"
        value="SN"
      >
        Senegal
      </option>
      <option
        key="RS"
        value="RS"
      >
        Serbia
      </option>
      <option
        key="SC"
        value="SC"
      >
        Seychelles
      </option>
      <option
        key="SL"
        value="SL"
      >
        Sierra Leone
      </option>
      <option
        key="SG"
        value="SG"
      >
        Singapore
      </option>
      <option
        key="SX"
        value="SX"
      >
        Sint Maarten
      </option>
      <option
        key="SK"
        value="SK"
      >
        Slovakia
      </option>
      <option
        key="SI"
        value="SI"
      >
        Slovenia
      </option>
      <option
        key="SB"
        value="SB"
      >
        Solomon Islands
      </option>
      <option
        key="SO"
        value="SO"
      >
        Somalia
      </option>
      <option
        key="ZA"
        value="ZA"
      >
        South Africa
      </option>
      <option
        key="GS"
        value="GS"
      >
        South Georgia and the South Sandwich Islands
      </option>
      <option
        key="KR"
        value="KR"
      >
        South Korea
      </option>
      <option
        key="SS"
        value="SS"
      >
        South Sudan
      </option>
      <option
        key="ES"
        value="ES"
      >
        Spain
      </option>
      <option
        key="LK"
        value="LK"
      >
        Sri Lanka
      </option>
      <option
        key="SD"
        value="SD"
      >
        Sudan
      </option>
      <option
        key="SR"
        value="SR"
      >
        Suriname
      </option>
      <option
        key="SJ"
        value="SJ"
      >
        Svalbard and Jan Mayen
      </option>
      <option
        key="SZ"
        value="SZ"
      >
        Swaziland
      </option>
      <option
        key="SE"
        value="SE"
      >
        Sweden
      </option>
      <option
        key="CH"
        value="CH"
      >
        Switzerland
      </option>
      <option
        key="SY"
        value="SY"
      >
        Syria
      </option>
      <option
        key="TW"
        value="TW"
      >
        Taiwan
      </option>
      <option
        key="TJ"
        value="TJ"
      >
        Tajikistan
      </option>
      <option
        key="TZ"
        value="TZ"
      >
        Tanzania
      </option>
      <option
        key="TH"
        value="TH"
      >
        Thailand
      </option>
      <option
        key="TG"
        value="TG"
      >
        Togo
      </option>
      <option
        key="TK"
        value="TK"
      >
        Tokelau
      </option>
      <option
        key="TO"
        value="TO"
      >
        Tonga
      </option>
      <option
        key="TT"
        value="TT"
      >
        Trinidad and Tobago
      </option>
      <option
        key="TN"
        value="TN"
      >
        Tunisia
      </option>
      <option
        key="TR"
        value="TR"
      >
        Turkey
      </option>
      <option
        key="TM"
        value="TM"
      >
        Turkmenistan
      </option>
      <option
        key="TC"
        value="TC"
      >
        Turks and Caicos Islands
      </option>
      <option
        key="TV"
        value="TV"
      >
        Tuvalu
      </option>
      <option
        key="VI"
        value="VI"
      >
        U.S. Virgin Islands
      </option>
      <option
        key="UG"
        value="UG"
      >
        Uganda
      </option>
      <option
        key="UA"
        value="UA"
      >
        Ukraine
      </option>
      <option
        key="AE"
        value="AE"
      >
        United Arab Emirates
      </option>
      <option
        key="GB"
        value="GB"
      >
        United Kingdom
      </option>
      <option
        key="US"
        value="US"
      >
        United States
      </option>
      <option
        key="UM"
        value="UM"
      >
        United States Minor Outlying Islands
      </option>
      <option
        key="UY"
        value="UY"
      >
        Uruguay
      </option>
      <option
        key="UZ"
        value="UZ"
      >
        Uzbekistan
      </option>
      <option
        key="VU"
        value="VU"
      >
        Vanuatu
      </option>
      <option
        key="VA"
        value="VA"
      >
        Vatican
      </option>
      <option
        key="VE"
        value="VE"
      >
        Venezuela
      </option>
      <option
        key="VN"
        value="VN"
      >
        Vietnam
      </option>
      <option
        key="WF"
        value="WF"
      >
        Wallis and Futuna
      </option>
      <option
        key="EH"
        value="EH"
      >
        Western Sahara
      </option>
      <option
        key="YE"
        value="YE"
      >
        Yemen
      </option>
      <option
        key="ZM"
        value="ZM"
      >
        Zambia
      </option>
      <option
        key="ZW"
        value="ZW"
      >
        Zimbabwe
      </option>
    </select>
  </td>
  <td
    className="is-hidden"
  >
    Saved
  </td>
  <td
    className="is-narrow"
  >
    <div
      className="is-hidden"
    >
      <button
        className="button is-primary"
        onClick={[Function]}
      >
        Save
      </button>
      <button
        className="button is-danger"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
    <button
      className="button is-link is-outlined is-hidden-touch"
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      className="button is-danger is-outlined"
      onClick={[Function]}
    >
      Ã
    </button>
  </td>
</tr>
`;
